File: _office365-lists-sdk/office365-base-sdk/office365-base-sdk/BaseEntity.m
//
//  BaseEntity.m
//  office365-base-sdk
//
//  Created by Gustavo on 7/21/14.
//  Copyright (c) 2014 Lagash. All rights reserved.
//

#import "BaseEntity.h"

@interface BaseEntity ()
@property NSDictionary *jsonData;
@end

@implementation BaseEntity

- (id) initWithJson:(NSData *)jsonData{
    NSDictionary *jsonResult = [NSJSONSerialization JSONObjectWithData:jsonData
                                                               options: NSJSONReadingMutableContainers
                                                                 error:nil];
    
    NSDictionary *jsonItem =[jsonResult valueForKey : @"d"];
    
    if(jsonItem == nil){
        return nil;
    }
    
    return [self initWithDictionary:jsonItem];
}

- (id) initWithDictionary:(NSDictionary *)dictionary{

    [self createFromJson:dictionary];
    return self;
}


-(void) createMetadata : (NSDictionary *) data{
    self.Metadata = [[Metadata alloc] initWith:data];
}

- (BaseEntity *)createFromJson:(NSDictionary *)data{
    NSDictionary *metadata = [data valueForKey : @"__metadata"];
    
    [self createMetadata : metadata];

    self.Id =[data valueForKey: @"Id"];
    self.jsonData = data;

    return self;
}

-(NSObject*) getData:(NSString *)name{
    return [self.jsonData valueForKey:name];
}

-(NSDictionary*) getRawResults{
    return self.jsonData;
}

@end

File: Sdk/office365_base_sdk/office365_base_sdk/BaseEntity.swift
//
//  BaseEntity.swift
//  office365-base-sdk
//
//  Created by Lagash on 6/17/14.
//  Copyright (c) 2014 microsoft. All rights reserved.
//

import Foundation

class BaseEntity: NSObject{

    var Id : String = "";
    
    var Metadata : MetaData = MetaData();
    var Properties : NSMutableDictionary = NSMutableDictionary();
    
    func createFromJson(data : NSDictionary) -> BaseEntity{
        return self;
    }
    
    func getPropertyValue(name : NSString) -> NSString{
        return self.Properties.valueForKey(name) as NSString;
    }
    
    func setAditionalProperties(properties : NSArray, data : NSDictionary){
        for property : AnyObject in properties{
            var stringProperty = property as NSString;
            self.Properties.setObject(data.valueForKey(stringProperty), forKey: stringProperty);
        }
    }
}

class MetaData : NSObject{

    var Id : String = "";
    var Uri : String = "";
    var type : String = "";
}